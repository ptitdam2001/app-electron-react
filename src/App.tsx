import React from 'react';
import './App.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Home } from './pages/Home';
import { Feed } from './pages/Feed';
import { Menu } from './features/core/Menu/Menu.component';
import { MenuContent } from './features/core/Menu/MenuContent.type';

const menuItems: MenuContent[] = [
  {
    path: '/',
    label: 'Home',
  },
  {
    path: '/feed',
    label: 'Feed',
  }
];

function App() {
  React.useEffect(() => {
    new Notification('Hello!', { body: 'This is a notification'});
  }, []);

  return (
    <div className="App">
      <Router>
        <header>
          <Menu items={menuItems} />
        </header>
        <section>
          <Switch>
            <Route path="/feed" component={Feed} />
            <Route component={Home} />
          </Switch>
        </section>
      </Router>
    </div>
  );
}

export default App;
