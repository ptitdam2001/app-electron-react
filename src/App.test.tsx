import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  const {container} = render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();

  expect(container.firstChild).toMatchSnapshot()
});
