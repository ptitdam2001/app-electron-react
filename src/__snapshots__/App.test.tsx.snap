// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  class="App"
>
  <header>
    <div
      class="MuiPaper-root MuiPaper-elevation1"
    >
      <div
        class="MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed"
          style="overflow: hidden;"
        >
          <div
            class="MuiTabs-flexContainer"
            role="tablist"
          >
            <a
              aria-disabled="false"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
              href="/"
              role="tab"
              tabindex="0"
            >
              <span
                class="MuiTab-wrapper"
              >
                Home
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
              href="/feed"
              role="tab"
              tabindex="-1"
            >
              <span
                class="MuiTab-wrapper"
              >
                Feed
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
          <span
            class="PrivateTabIndicator-root-1 PrivateTabIndicator-colorPrimary-2 MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
  </header>
  <section>
    <div>
      Home page
    </div>
  </section>
</div>
`;
